lib/network/modules/components/shared/Label.js

```diff
@@ -178,141 +202,178 @@ class Label {
-        ;
-
-        // 'color' follows the standard flow
-         modOptions.color =
-          getP(options     ,  mod, 'color') ||
-          getP(groupOptions,  mod, 'color') ||
-          getP(groupOptions, null, 'color') ||
-          this.fontOptions.color
-        ;
-
-        // 'mode' follows the standard flow
-        modOptions.mod =
-          getP(options     ,  mod, 'mod') ||
-          getP(groupOptions,  mod, 'mod') ||
-          getP(groupOptions, null, 'mod') ||
-          modDefaults.mod
-        ;
-
-
-        // It's important that we size up defaults similarly if we're
-        // using default faces unless overriden. We want to preserve the
-        // ratios closely - but if faces have changed, all bets are off.
-        let ratio;
-
-        // NOTE: Following condition always fails, because modDefaults
-        //       has no explicit font property. This is deliberate, see
-        //       var's 'NodesHandler.defaultOptions.font[mod]'.
-        //       However, I want to keep the original logic while refactoring;
-        //       it appears to be working fine even if ratio is never set.
-        // TODO: examine if this is a bug, fix if necessary.
-        //
-        if ((modOptions.face === modDefaults.face) &&
-            (this.fontOptions.face === defaultOptions.font.face)) {
-
-          ratio = this.fontOptions.size / Number(defaultOptions.font.size);
+          if (multiFont.hasOwnProperty(option)) {
+            return multiFont[option];
+          }
         }
+      }
 
+      // option not mentioned in the multi font options; take it from the parent font options
+      // These have already been converted with getBasicOptions(), so use the converted values
+      if (this.fontOptions.hasOwnProperty(option)) {
+        return this.fontOptions[option];
+      }
 
-        modOptions.size =
-          getP(options     ,  mod, 'size')        ||
-          getP(groupOptions,  mod, 'size')        ||
-          (ratio? modDefaults.size * ratio: null) || //   Scale the mod size using the same ratio
-          getP(groupOptions, null, 'size')        ||
-          this.fontOptions.size
-        ;
+      // A value **must** be found; you should never get here.
+      throw new Error("Did not find value for multi-font for property: '" + option + "'");
+    };
 
-        modOptions.vadjust =
-          getP(options     , mod, 'vadjust')                     ||
-          getP(groupOptions, mod, 'vadjust')                     ||
-          (ratio? modDefaults.vadjust * Math.round(ratio): null) || // Scale it using the same ratio
-          this.fontOptions.vadjust
-        ;
 
+   /**
+    * Return all options values for the given multi-font.
+    *
+    * All available option objects are trawled in the set order to construct the option values.
+    *
+    * @param {MultiName} multiName sub path for the mod-font
+    * @returns {ModOptions}
+    */
+    var getFontOptions = (multiName) => {
```

Comment:
Why do you use `var`?  Why not use `const` and `let`? 
(for consistency)

Reply:
A quick response on this one in passing:

I prefer `var` because chrome has issues dealing with `const` and `let`, it can't seem to display the values correctly inline (usually shows `undefined` even though it has a value). I find this bothersome to the point of forsaking them, even though `let` would be better.

If you have a solution for this, let me know.

Reply:
But it seems that you douse `let` in the rest of the file. How does that work?

Reply:
I really try to use `let`. But if it gets in the way of debugging, I revert it to `var`.

Reply:
Oh well, I can change this since I'm done with debugging.